### DISCUSSION TITLE
set vm.args in kubernetes

### DISCUSSION THREAD

**Discussion by 1995parham** [emqx/emqx#6924] [Q&A] [answered]
Is there any way for setting `vm.args` with emqx helm chart with environment variables? For example, I want to set `--cpus` to have container friendly environment or increase the number of `+IOt` for handling more concurrent connections.

**Comment by zmstone** [comment, marked as answer]
Hi @1995parham 

in EMQ X 4.3, only the below args have a mapping rule from emqx.conf (hence environment variable)

`grep mapping priv/emqx.schema | grep vm_args`
```
{mapping, "node.name", "vm_args.-name", [
{mapping, "node.ssl_dist_optfile", "vm_args.-ssl_dist_optfile", [
{mapping, "node.cookie", "vm_args.-setcookie", [
{mapping, "node.heartbeat", "vm_args.-heart", [
{mapping, "node.async_threads", "vm_args.+A", [
{mapping, "node.process_limit", "vm_args.+P", [
{mapping, "node.max_ports", "vm_args.+Q", [
{mapping, "node.dist_buffer_size", "vm_args.+zdbbl", [
{mapping, "node.fullsweep_after", "vm_args.-env ERL_FULLSWEEP_AFTER", [
{mapping, "node.crash_dump", "vm_args.-env ERL_CRASH_DUMP", [
{mapping, "node.dist_net_ticktime", "vm_args.-kernel net_ticktime", [
```

e.g. `export EMQX_node__process_limit=100000000`

but not for others.

`+IOt` is unfortunately not parameterised by `emqx.schema`

**Reply by 1995parham** [reply]
@zmstone thanks for your answer. Is it okay to create a PR for this on the main-v4.3 branch? or you don't accept any PR on this branch?

**Reply by zmstone** [reply]
Pr is welcome.
